hummingbot/strategy/pure_market_making/pure_market_making.pyx

```diff
@@ -1083,9 +1076,20 @@ cdef class PureMarketMakingStrategy(StrategyBase):
                 return False
         return True
 
-    # Cancel active non hanging orders
-    # Return value: whether order cancellation is deferred.
+    cdef c_cancel_active_orders_on_max_age_limit(self):
+        """
+        Cancels active non hanging orders if they are older than max age limit
+        """
+        cdef:
+            list active_orders = self.active_non_hanging_orders
+        if active_orders and any(order_age(o) > self._max_order_age for o in active_orders):
+            for order in active_orders:
```

Comment:
Is the intention to cancel all active orders? Shouldn't the logic cancel only the orders for which age is > _max_order_age?